ui/imports/shared/views/AssetsView.qml

```diff
@@ -337,6 +340,12 @@ ColumnLayout {
                 text: qsTr("Receive")
                 onTriggered: root.receiveRequested(symbol)
             }
+            StatusAction {
+                icon.name: "swap"
+                text: qsTr("Swap")
+                enabled: Global.featureFlags.swapEnabled
```

Comment:
I thought it was decided against using `Global` for the feature flags?

Reply:
I am not sure, was done by the wallet connect team before us and Ive used the same here. Can you elaborate what was the approach decided then?

Reply:
The discussion ended with the [comment here](https://github.com/status-im/status-desktop/pull/14590#issuecomment-2104354449). Basically the usage of `Global` was a shortcut from my side to have it working without building a proper understanding of dependencies needed to create and share a specialized `FeatureFlagsStore` with all the other components (`DappsStore` and `SwapStore`) while being available globally so you can access it at any level to guard the feature code. I was basically leaving the proper implementation for when we finish the high priority dApps tickets or the UI team has bandwidth to implement it.

Reply:
ok so is there a task created to do that? so that its not lost? then the same can be done for swap in that task too.

Reply:
> ok so is there a task created to do that? so that its not lost? then the same can be done for swap in that task too

Yes, here it is https://github.com/status-im/status-desktop/issues/14697